core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -23,377 +22,271 @@ import scala.jdk.CollectionConverters._
-        extractSpendAggregation(rowsForGoogleProjectId,
-                                currency,
-                                aggregationKey = SpendReportingAggregationKeyWithSub(key),
-                                workspaceProjectsToNames = workspaceProjectsToNames
-        )
-      }
-      val workspaceName = workspaceProjectsToNames.getOrElse(
-        googleProjectId,
-        throw new RawlsExceptionWithErrorReport(
-          ErrorReport(StatusCodes.BadGateway, s"unexpected project ${googleProjectId.value} returned by BigQuery")
-        )
+  def getSpendExportConfiguration(project: RawlsBillingProjectName): Future[BillingProjectSpendExport] = dataSource
+    .inTransaction(_.rawlsBillingProjectQuery.getBillingProjectSpendConfiguration(project))
+    .recover { case _: RawlsException =>
+      throw RawlsExceptionWithErrorReport(
+        StatusCodes.BadRequest,
+        s"billing account not found on billing project ${project.value}"
       )
-      SpendReportingForDateRange(
-        cost.toString(),
-        credits.toString(),
-        currency.getCurrencyCode,
-        workspace = Option(workspaceName),
-        googleProjectId = Option(googleProjectId),
-        subAggregation = subAggregation
+    }
+    .map {
+      _.getOrElse(
+        throw RawlsExceptionWithErrorReport(StatusCodes.NotFound, s"billing project ${project.value} not found")
       )
-    }.toList
-
-    SpendReportingAggregation(
-      SpendReportingAggregationKeys.Workspace,
-      workspaceSpend
-    )
-  }
+    }
 
-  private def extractCategorySpendAggregation(rows: List[FieldValueList],
-                                              currency: Currency,
-                                              subAggregationKey: Option[SpendReportingAggregationKey] = None,
-                                              workspaceProjectsToNames: Map[GoogleProject, WorkspaceName] = Map.empty
-  ): SpendReportingAggregation = {
-    val spendByCategory: Map[TerraSpendCategory, List[FieldValueList]] =
-      rows.groupBy(row => TerraSpendCategories.categorize(row.get("service").getStringValue))
-    val categorySpend = spendByCategory.map { case (category, rowsForCategory) =>
-      val (cost, credits) = sumCostsAndCredits(rowsForCategory, currency)
-      val subAggregation = subAggregationKey.map { key =>
-        extractSpendAggregation(rowsForCategory,
-                                currency,
-                                aggregationKey = SpendReportingAggregationKeyWithSub(key),
-                                workspaceProjectsToNames = workspaceProjectsToNames
-        )
-      }
-      SpendReportingForDateRange(
-        cost.toString,
-        credits.toString,
-        currency.getCurrencyCode,
-        category = Option(category),
-        subAggregation = subAggregation
-      )
-    }.toList
+  def getWorkspaceGoogleProjects(projectName: RawlsBillingProjectName): Future[Map[String, WorkspaceName]] =
```

Comment:
It's arguable that it would have been better to leave the map key as `GoogleProject` instead of a `String`.
I was getting frustrated with all the verbose types floating around, and the bloat that was added by the having to constantly unbox something that was essentially a renamed String. 
If it wasn't already a separate case class, I'd probably just add `type GoogleProject = String`, which would give the clarity of what we were tossing around, without needing to have the (mental) overhead of the separate class.